build-tools/xaprepare/xaprepare/Application/Context.cs

```diff
@@ -329,6 +329,17 @@ partial class Context : AppObject
 		/// </summary>
 		public RefreshableComponent ComponentsToRefresh { get; set; }
 
+		/// <summary>
+		///   True if the Azure Pipelines build was automatically triggered from a pull request.
+		///   Manually triggered pull request builds will have a 'Manual' build reason, and return false.
+		/// </summary>
+		public bool BuildingAzurePullRequest {
+			get {
+				string buildReason = Environment.GetEnvironmentVariable ("BUILD_REASON");
+				return !String.IsNullOrEmpty (buildReason) && buildReason == "PullRequest";
+			}
```

Comment:
Should probably use case-insensitive string comparisons for environment variables, just in case someone decides to manually override...